### PR TITLE
JP-1923 tsgrism and wfss coordinate frames improvement

### PR DESCRIPTION
**Pull request by nden**
Resolves [JP-1923](https://jira.stsci.edu/browse/JP-1923)/#5746

Added a spectral axis to coordinate frames where necessary.
Cleaned code.

### CODE REVIEW

**Review by hbushouse — APPROVED** [review]
Looks good to me. Needs a change log entry.

One question: will the change in frame names require changes to downstream code, e.g. the stuff being written by DA folks to do WFSS analysis? Should inform them if this will affect them.

File: jwst/assign_wcs/nircam.py [hunk 1]
```diff
@@ -410,15 +406,43 @@ def wfss(input_model, reference_files):
 
     # pass the x0,y0, wave, order, through the pipeline
     imagepipe = []
-    world = image_pipeline.pop()
+    world = image_pipeline.pop()[0]
```
Comment (mcara) [inline jwst/assign_wcs/nircam.py:409]: Wouldn't this cause `DeprecationWarning`?
```suggestion
    world = image_pipeline.pop().frame
```

File: jwst/assign_wcs/niriss.py [hunk 2]
```diff
@@ -445,11 +446,15 @@ def wfss(input_model, reference_files):
     # so the user doesn't have to
 
     imagepipe = []
-    world = image_pipeline.pop()
+    world = image_pipeline.pop()[0]
```
Comment (mcara) [inline jwst/assign_wcs/niriss.py:449]:
```suggestion
    world = image_pipeline.pop().frame
```

**Review by mcara — APPROVED** [review]
One minor comment on indexing pipeline and add changelog.

File: jwst/assign_wcs/nircam.py (hunk 1 above)
Comment (nden) [reply jwst/assign_wcs/nircam.py:409]: Ah, after the `pop` it's just a tuple.

Comment (mcara) [reply jwst/assign_wcs/nircam.py:409]: That's unexpected (to me)

**Pull request merged**